### ISSUE TITLE
needs cabal-install-3.2 to build

### ISSUE THREAD

**Issue by juhp** [phadej/cabal-extras#23]
I pondered opening a PR against README.md, but anyway just noting that I could only build cabal-diff with cabal-install-3.2 (I tried with both ghc-8.6.5 and ghc-8.8.3), otherwise cabal-install (2.4.1 and 3.0) couldn't resolve the dependencies. :)

cabal-diff is very cool, thanks!!

**Comment by fendor** [comment]
For me building cabal-extras stuff with `cabal-install 3.0` with `ghc-8.6.5` on Win 10 and NIXOS 20.04 succeeds.

**Comment by juhp** [comment]
Just for the record this is what I get with ghc-8.6.5 and cabal-install-3.0:

```
Resolving dependencies...
cabal-3.0: Could not resolve dependencies:
[__0] trying: base-4.12.0.0/installed-4.1... (user goal)
[__1] trying: binary-0.8.8.0 (user goal)
[__2] next goal: ghc (user goal)
[__2] rejecting: ghc-8.6.5/installed-8.6... (conflict: binary==0.8.8.0, ghc =>
binary==0.8.6.0/installed-0.8...)
[__2] trying: ghc-8.6.5
[__3] rejecting: ghc:-buildable (conflict: base==4.12.0.0/installed-4.1...,
ghc -buildable => base<0)
[__3] rejecting: ghc:+buildable (manual flag can only be changed explicitly)
[__3] fail (backjumping, conflict set: base, ghc, ghc:buildable)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: binary, base, ghc, ghc:buildable
Try running with --minimize-conflict-set to improve the error message.
```

Anyway feel free to close this if it is not helpful.
I can build fine with 3.2 anyway :+1:

**Comment by phadej** [comment]
@juhp, please do, otherwise this seems to be non-reproducible issue.

**Comment by juhp** [comment]
I edited and updated my comment.

**Comment by phadej** [comment]
@juhp

```
trying: binary-0.8.8.0 (user goal)
```

is suspicious, where from it comes? Do you have `binary-0.8.8.0` in `~/.ghc/x86_64-linux-8.6.5/environments/default` (or similar for GHC-8.6.5). Then I think it's a https://github.com/haskell/cabal/issues/6410 which is fixed in cabal-install-3.0.1.0 (which is unreleased) and cabal-install-3.2.

**Comment by juhp** [comment]
> ```
> trying: binary-0.8.8.0 (user goal)
> ```
> is suspicious, where from it comes? Do you have `binary-0.8.8.0` in `~/.ghc/x86_64-linux-8.6.5/environments/default` (or similar for GHC-8.6.5). Then I think it's a [haskell/cabal#6410](https://github.com/haskell/cabal/issues/6410) which is fixed in cabal-install-3.0.1.0 (which is unreleased) and cabal-install-3.2.

I suspect you are exactly right - not sure how binary-0.8.8.0 would get into my default environment..? (Apparently I also had a newer version of text there).

Anyway thank you!
